processor/resourcedetectionprocessor/internal/consul/config.go

```diff
@@ -0,0 +1,41 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consul
+
+import (
+	"go.opentelemetry.io/collector/config"
+)
+
+type Config struct {
+	config.ProcessorSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+
+	// Address is the address of the Consul server
+	Address string `mapstructure:"address"`
+
+	// Datacenter to use. If not provided, the default agent datacenter is used.
+	Datacenter string `mapstructure:"datacenter"`
+
+	// Token is used to provide a per-request ACL token
```

Comment:
What is the agent's default token?

Reply:
by default, consul doesn't authenticate requests and don't have any default token

Reply:
Let's remove the sentence about the agent's default token then, sounds like there isn't one

Reply:
wanted to keep the general language since that's how its documented in the consul API (https://github.com/hashicorp/consul/blob/3949acbc1d9646b1aa536b0c61736fda7e737f6f/api/api.go#L152-L154), but added a call out about when its required